tests/displayBooksListTest.php

```diff
@@ -47,10 +48,79 @@ public function test_success_display_books_list()
         $this->assertEquals($expectedOutput, $actualOutput);
     }
 
-    public function test_empty_array()
+
+
+
+    public function testDisplayBooksListWithValidBooks()
     {
+        // Mock an array of Book objects
+        $books = [
+            new Book(
+                1,
+                'Title',
+                1,
+                'AuthorName1',
+                'https://wikipedia.org',
+                1234567890123,
+                'Summary',
+                1900,
+                1,
+                'Spanish',
+                'https://wikipedia.org',
+                'https://wikipedia.org',
+                0
+            ),
+            new Book(
+                1,
+                'Title',
+                1,
+                'AuthorName2',
+                'https://wikipedia.org',
+                1234567890123,
+                'Summary',
+                1900,
+                1,
+                'Spanish',
+                'https://wikipedia.org',
+                'https://wikipedia.org',
+                0
+            )
+        ];
```

Comment:
Same here, params are causing a type error.

TypeError: Collection\Entities\Book::__construct(): Argument #2 ($author_id) must be of type int, string given